documents/coredocument.go

```diff
@@ -203,32 +206,26 @@ func (cd *CoreDocument) PrepareNewVersion(collaborators []string, initSalts bool
 	return ncd, nil
 }
 
-// addCollaboratorsToReadSignRules adds the given collaborators to a new read rule with READ_SIGN capability.
-// The operation is no-op if no collaborators is provided.
-// The operation is not idempotent. So calling twice with same accounts will lead to read rules duplication.
-func (cd *CoreDocument) addCollaboratorsToReadSignRules(collaborators []identity.DID) {
+// newRole returns a new role with random role key
+func newRole() *coredocumentpb.Role {
+	return &coredocumentpb.Role{RoleKey: utils.RandomSlice(idSize)}
+}
+
+// addCollaboratorsToNewRole creates a new Role and adds the given collaborators to this Role.
+// The Role is then returned.
+// The operation returns a nil Role if no collaborators are provided.
+func (cd *CoreDocument) addCollaboratorsToNewRole(collaborators []identity.DID) *coredocumentpb.Role {
```

Comment:
This is apure function. Lets not make it part of the COreDocument. I would prefer it if you create a new function called `newRoleWithCollaborators` instead